### DISCUSSION TITLE
Start Astro programatically

### DISCUSSION THREAD

**Discussion by screendriver** [withastro/roadmap#357] [Proposal]
It would be nice when Astro would have an API like `Vite` to interact with it programatically. For example I wish I could start / create the dev server like in `Vite` with [createServer()](https://vitejs.dev/guide/api-javascript.html#createserver). This would enable us to have tighter integration with overall development processes like for example booting up another test server first and then pass the URL of that server into `Astro` as environment variable to fetch data from it via local API calls or create temporary files and so on.

Right now I have to use [zx](https://github.com/google/zx) to write a script that does that and in the end do a

```
await $`npx astro dev`;
```

**Comment by florian-lefebvre** [comment]
Closing because this is outdated

**Reply by delucis** [reply]
Adding that this is has been supported for a while. Docs for the programmatic API are here: https://docs.astro.build/en/reference/programmatic-reference/
